shared/constants/transaction.js

```diff
@@ -0,0 +1,95 @@
+// 1. `'unapproved'` - A new transaction that the user has not approved or rejected
+// 2. `'approved'`   - The user has approved the transaction in the MetaMask UI
+// 3. `'rejected'`   - The user has rejected the transaction in the MetaMask UI
+// 4. `'signed'`     - The transaction has been signed
+// 5. `'submitted'`  - The transaction has been submitted
+// 6. `'failed'`     - The transaction has failed for some reason
+// 7. `'dropped'`    - The transaction was dropped due to a tx with same nonce being accepted
+export const TRANSACTION_STATUS_APPROVED = 'approved'
+export const TRANSACTION_STATUS_CONFIRMED = 'confirmed'
+export const TRANSACTION_STATUS_UNAPPROVED = 'unapproved'
+export const TRANSACTION_STATUS_REJECTED = 'rejected'
+export const TRANSACTION_STATUS_SIGNED = 'signed'
+export const TRANSACTION_STATUS_SUBMITTED = 'submitted'
+export const TRANSACTION_STATUS_FAILED = 'failed'
+export const TRANSACTION_STATUS_DROPPED = 'dropped'
+
+// Transaction Category is a MetaMask construct used internally
+// Used to categorize transactions by purpose for displaying information to the user
+// 1. `'tranfser'`            - A token transaction where the user is sending tokens that they
+//                              own to another address
+// 2. `'transferfrom'`        - A token transaction transferring tokens from an account that
+//                              the sender has an allowance of (see the approve category).
+// 3. `'approve'`             - A token transaction requesting an allowance of the token to
+//                              spend on behalf of the user
+// 4. `'incoming'`            - An incoming (deposit) transaction
+// 5. `'sentEther'`           - A transaction sending ether to a recipient
+// 6. `'contractInteraction'` - A transaction that is interacting with a smart contract's methods
+// 7. `'contractDeployment'`  - A transaction that deployed a smart contract
+export const TRANSACTION_CATEGORY_TOKEN_METHOD_TRANSFER = 'transfer'
+export const TRANSACTION_CATEGORY_TOKEN_METHOD_APPROVE = 'approve'
+export const TRANSACTION_CATEGORY_TOKEN_METHOD_TRANSFER_FROM = 'transferfrom'
+export const TRANSACTION_CATEGORY_INCOMING = 'incoming'
+export const TRANSACTION_CATEGORY_SENT_ETHER = 'sentEther'
+export const TRANSACTION_CATEGORY_CONTRACT_INTERACTION = 'contractInteraction'
+export const TRANSACTION_CATEGORY_DEPLOY_CONTRACT = 'contractDeployment'
+
+// Transaction Group Status is a MetaMask construct to track status of groups
+// of transactions.
+// 1. `'cancelled'` - A cancel type transaction in the group was confirmed
+// 2. `'pending'`   - The primaryTransaction of the group has a status that is one of:
+//                    1. TRANSACTION_STATUS_APPROVED
+//                    2. TRANSACTION_STATUS_UNAPPROVED
+//                    3. TRANSACTION_STATUS_SUBMITTED
+export const TRANSACTION_GROUP_STATUS_CANCELLED = 'cancelled'
+export const TRANSACTION_GROUP_STATUS_PENDING = 'pending'
```

Comment:
Should "confirmed" and "failed" be represented here as well?

Reply:
I *could* but these are the statuses that don't exist as transaction statuses. Honestly i could go either way with it. If we had separate consts for those it would enable us to change the string value for one without changing the other in the future. These two cases here represent areas where we override the status by inferring from the transaction data. For example, pending happens when the transaction is in one of the mentioned statuses, and cancelled happens when a cancel type transaction has been confirmed... the others are just inherited from the primary transaction.